src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs

```diff
@@ -3631,5 +3631,407 @@ .maxstack  9
   IL_0070:  ret
 }");
         }
+
+        [Fact]
+        public void TwoStructClosures()
+        {
+            var source0 = MarkedSource(@"
+public class C 
+    {
+    public void F()            
+    {
+        int x = 0;
+        {
+            int y = 0;
+            // Captures two struct closures
+            int L() => x + y;
+        }
+    }
+}");
+
+            var source1 = MarkedSource(@"
+public class C 
+    {
+    public void F()            
+    {
+        int x = 0;
+        {
+            int y = 0;
+            // Captures two struct closures
+            int L() => x + y + 1;
+        }
+    }
+}");
+            var compilation0 = CreateStandardCompilation(source0.Tree, options: ComSafeDebugDll);
+            var compilation1 = compilation0.WithSource(source1.Tree);
+            var v0 = CompileAndVerify(compilation0);
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+
+            var f0 = compilation0.GetMember<MethodSymbol>("C.F");
+            var f1 = compilation1.GetMember<MethodSymbol>("C.F");
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(new SemanticEdit(SemanticEditKind.Update, f0, f1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            var md1 = diff1.GetMetadata();
+            var reader1 = md1.Reader;
```

Comment:
>ar reader1 = md1.Reader; [](start = 13, length = 24)

We should validate synthesized members using VerifySynthesizedMembers

Reply:
Thank you!

---
In reply to: [137682070](https://github.com/dotnet/roslyn/pull/21971#discussion_r137682070) [](ancestors = 137682070)